### PR TITLE
S4 finish game

### PR DESCRIPTION
**Pull request by kimberleyworld**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -9,8 +9,9 @@
     <script defer src="index.js"></script>
 </head>
 <body>
+<div class="modal-blur">
 <header>
```
Comment (charliecog) [inline index.html]: need indent

File: index.html [hunk 2]
```diff
@@ -37,9 +38,18 @@ <h2>Instructions:</h2>
...
 </header>
-<section id="hide">
+<section id="gameContainer">
     <h1 id="gameTitle">Locate the Lynx</h1>
     <div id="gridContainer"></div>
 </section>
+</div>
+
+    <div class="endModal">
+        <div class="endModalContent">
```
Comment (charliecog) [inline index.html]: dont need this extra div layer

File: index.html [hunk 3]
```diff
@@ -37,9 +38,18 @@ <h2>Instructions:</h2>
...
+            <h2 class="scoreMessage"></h2>
+            <button class="play-again">Play again!</button>
```
Comment (charliecog) [inline index.html]: could just use .endModal button in CSS instead

File: index.js [hunk 4]
```diff
@@ -81,18 +52,105 @@ function generateSquares(numLynx, miss) {
     })
 }
 
-function turnCard() {
+function turnCard(numLynx, miss) {
```
Comment (charliecog) [inline index.js]: miss could be more semantically named

File: index.js [hunk 5]
```diff
@@ -81,18 +52,105 @@ function generateSquares(numLynx, miss) {
...
         item.addEventListener('click', () => {
-            if(item.dataset.hit === '1') {
+            if (item.dataset.hit === '1') {
                 item.style.backgroundImage = "url('project-assets/babylynx2.jpg')"
                 item.textContent = 'Hooray, you\'ve found a lynx!'
-            } else {
+                hits++
+                item.dataset.hit = '2'
+                if (hits === numLynx) {
+                    modalDiv.style.display = "inline-block"
```
Comment (charliecog) [inline index.js]: whole block a potential function to be called with params passed in for the unique data?

File: index.js [hunk 6]
```diff
@@ -81,18 +52,105 @@ function generateSquares(numLynx, miss) {
...
-}
\ No newline at end of file
+}
+
+function newGame() {
+    //removes contents of the grid
+    document.getElementById('gridContainer').innerHTML = ''
+
+    //hide the game element
+    document.getElementById('gameContainer').style.display="none"
```
Comment (charliecog) [inline index.js]: space either side of equals

File: index.js [hunk 7]
```diff
@@ -81,18 +52,105 @@ function generateSquares(numLynx, miss) {
...
+
+    //reset form values to ''
+    document.getElementById('numLynx').value = ''
+    document.getElementById('gridSize').value = ''
+
+    //reshow splash screen
+    document.querySelector('#mainTitle').scrollIntoView({
```
Comment (charliecog) [inline index.js]: not sure this is doing anything

File: styles.css [hunk 8]
```diff
@@ -111,14 +112,75 @@ section {
...
 }
 
+.endModal {
+    display: none;
+    width: 70%;
+    height: 30%;
+    border: 2px solid gray;
+    text-align: center ;
+    padding: 10px;
+    position: absolute;
```
Comment (charliecog) [inline styles.css]: position fixed allows scroll

File: index.js [hunk 9]
```diff
@@ -81,18 +52,105 @@ function generateSquares(numLynx, miss) {
...
+        misses = 0
+        generateSquares(numLynx, miss)
+        turnCard(numLynx, miss)
+        //sets the grid container display to default
+        document.getElementById('gameContainer').style.display = "flex"
+        //scroll to top of game play area
+        document.querySelector('#gameContainer').scrollIntoView({
+            behavior: 'smooth'
+        })
+        document.querySelector('header').style.minHeight = "auto"
```
Comment (charliecog) [inline index.js]: remove this

Comment (charliecog) [reply index.js]: and remove resets elsewhere (cmd + f to find them)

File: styles.css [hunk 10]
```diff
@@ -111,14 +112,75 @@ section {
...
+    padding: 10px;
+    transition: 0.3s;
+    margin-top: 50px;
+    width: 50%;
+    position: relative;
+    bottom: 0;
+}
+
+.endModalContent button:hover {
+    background-color: #593837;
```
Comment (charliecog) [inline styles.css:163]: add in cursor pointer

File: styles.css [hunk 11]
```diff
@@ -111,14 +112,75 @@ section {
...
+div[data-hit ~= "2"]{
     color: transparent;
 }
 
+.endModal {
+    display: none;
+    width: 70%;
+    height: 30%;
+    border: 2px solid gray;
+    text-align: center ;
```
Comment (charliecog) [inline styles.css:129]: not sure this is doing anything

Comment (Reductito) [reply styles.css:129]: I think we do need it because if we change any one of the modal's CSS options it breaks

File: styles.css [hunk 12]
```diff
@@ -135,11 +189,16 @@ div[data-hit ~= "0"] {
...
 
-#gameTitle {
-    background-color: rgba(224,216,222, 0.5);
-    padding: 30px 40px;
-    border-radius: 5px;
-    text-align: center;
-}
+@media screen and (max-width: 410px) {
+    .scoreMessage {
+        font-size: 24px;
```
Comment (charliecog) [inline styles.css]: make this 20px and it will sort out the iphone 5 view

File: index.js [hunk 13]
```diff
@@ -81,18 +53,92 @@ function generateSquares(numLynx, miss) {
...
+    const numSnakes = gridSize - numLynx
+
+    //reset
+    document.getElementById('gridContainer').innerHTML = ''
+    document.getElementById( 'errorMessage').textContent = ''
+
+    //validates form input
+    if (!validateInput(gridSize, numLynx)) {
+        document.getElementById('errorMessage').textContent = errorMessage
+        document.getElementById('gameContainer').style.display = "none"} else {
```
Comment (charliecog) [inline index.js]: else should be formatted in the proper way eg: new line for the } else {

File: index.js [hunk 14]
```diff
@@ -81,18 +53,92 @@ function generateSquares(numLynx, miss) {
     })
 }
 
-function turnCard() {
+function decideOutcome(modalText) {
```
Comment (charliecog) [inline index.js]: modalText could use a default param that says "something went wrong"

**Review by charliecog — APPROVED** [review]

**Pull request merged**